### ISSUE TITLE
Training on qwen2

### ISSUE THREAD

**Issue by LUXINGYU23** [Wangningsen/2025CG-PJ#1]
This is a great job! I don't seem to be able to generate ideal answers even after training on qwen3. Has the author tried training on qwen2 ？

**Comment by Wangningsen** [comment]
> We want to be transparent about our training efforts. While we've seen success with some models, we hit a significant roadblock trying to fine-tune Qwen3 1.7B with the CAD-Recode v1.5 dataset to outperform the original CAD-Recode model (which uses Qwen2 1.5B). Despite tremendous effort and numerous training epochs, our fine-tuned Qwen3 model consistently failed to generate any correct answers on the Fusion360 test set.

 
We didn't try qwen2. It is quite strange that training qwen3 with cadrecode dataset **cannot enable it to generate even one** correct answer on fusion360 testset. 

One of my classmates chose this work as his baseline too, but he trained qwen3 with one of the training set of the tested datasets in the original paper. In that way his model can generate correct answer and surpass the baseline's performance on testset. Maybe you can give it a try.

By the way, the lora codes here have some errors. The MLP mapping layer is not correctly saved after training. We didn't realize this serious error until we were near the ddl. Thus we directly adopt the MLP mapping layer from the original model weight and it works fine. Seemly that part is greatly changed after training.

**Comment by LUXINGYU23** [comment]
Thanks so much for sharing these detailed insights!💕

**Comment by Wangningsen** [comment]
@LUXINGYU23 111 bro where is your pr

**Comment by LUXINGYU23** [comment]
> [@LUXINGYU23](https://github.com/LUXINGYU23) 111 bro where is your pr
Here, because our engineering structures are slightly different, in order to save time, I have directly opened a new repository to store my complete code.
https://github.com/LUXINGYU23/OpenCADRecode

**Comment by Wangningsen** [comment]
> > [@LUXINGYU23](https://github.com/LUXINGYU23) 111 bro where is your pr
> > Here, because our engineering structures are slightly different, in order to save time, I have directly opened a new repository to store my complete code.
> > https://github.com/LUXINGYU23/OpenCADRecode

Nice work.
